### PR TITLE
Support deleting runs from the Flows service

### PR DESCRIPTION
**Pull request by kurtmckee**
* Support deleting runs from the Flows service.

The SDK API documentation was built and verified. The link to the Flows OpenAPI page will be verified once the associated PR is merged and the OpenAPI docs are rebuilt and published.

----
:books: Documentation preview :books:: https://globus-sdk-python--745.org.readthedocs.build/en/745/

### CODE REVIEW

File: src/globus_sdk/services/flows/client.py [hunk 1]
```diff
@@ -668,6 +668,38 @@ def update_run(
 
         return self.put(f"/runs/{run_id}", data=data)
 
+    def delete_run(self, run_id: UUIDLike) -> GlobusHTTPResponse:
+        """
+        Delete a run.
+
+        :param run_id: The ID of the run to update
```
Comment (ada-globus) [inline src/globus_sdk/services/flows/client.py:675]:
```suggestion
        :param run_id: The ID of the run to delete
```

**Review by ada-globus — APPROVED** [review]
Looks great! One minor note in the documentation.

File: src/globus_sdk/services/flows/client.py [hunk 2]
```diff
@@ -668,6 +668,38 @@ def update_run(
...
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: flows.delete_run
+
+            .. tab-item:: API Info
+
+                .. extdoclink:: Delete Run
+                    :service: flows
+                    :ref: Runs/paths/~1runs~1{run_id}~1release/post
```
Comment (ada-globus) [inline src/globus_sdk/services/flows/client.py:698]: Caught this after approving, looks like this is not rendering right in the docs. When I click the preview it takes me to: https://globusonline.github.io/flows/#tag/Runs/paths/~1runs~1%7Brun_id%7D~1release/post

but the actual location is: https://globusonline.github.io/flows/#tag/Runs/paths/~1runs~1{action_id}~1release/post

I know that the `run_id` change will work out as soon as the Flows OpenAPI docs are redeployed, but I think the urlencoding of the curly braces might still be broken (tested this manually in the browser by replacing with `action_id` to go to the current location but leaving the urlencoded entities and it didn't seem to resolve correctly for me).

File: src/globus_sdk/services/flows/client.py (hunk 1 above)
Comment (kurtmckee) [reply src/globus_sdk/services/flows/client.py]: This should be addressed now.

**Review by sirosen — APPROVED** [review]
Thanks for updating the changelog per our conversation!

Everything here looks good, and I especially appreciate that you included a use of `expandtestfixture` in the docs. I'd like to get that on _all_ of our API methods at some point.

**Pull request merged**